arch/x86/core/intel64/cpu.c

```diff
@@ -144,14 +144,14 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	uint8_t vector = ((unsigned long) x86_ap_start) >> 12;
 	uint8_t apic_id;
 
-#ifdef CONFIG_ACPI
-	struct acpi_madt_local_apic *lapic = acpi_local_apic_get(cpu_num);
-
-	if (lapic != NULL) {
-		/* We update the apic_id, __start will need it. */
-		x86_cpu_loapics[cpu_num] = lapic->Id;
-	}
-#endif
+	IF_ENABLED(CONFIG_ACPI, ({
```

Comment:
If ACPICA module not included in west this can fail ?

Reply:
also #include <acpi.h> is under CONFIG_ACPI

Reply:
> If ACPICA module not included in west this can fail ?

No, it will still work (as you can also see from CI passing). There is also no problem with the #ifdef guard for acpi.h. IF_ENABLED works the same way as #ifdef, i.e. it's not the same as IS_ENABLED.

For the future, I'm thinking that maybe we should split Zephyr's acpi.h into two separate headers: a public and an internal one. The public one would not depend on any ACPI definitions, so this would be the header that any c-file would use that's not 100% ACPI related (like this cpu.c). We would however need to create some new wrappers/abstractions to allow access to all necessary functionality this way.